docs/example-terraform.md

```diff
@@ -0,0 +1,187 @@
+
+```json5
+local cloudbuild = import "cloudbuild.jsonnet";
+...
+{
+  "output.tf": output,
+  "provider.tf": provider,
+  [if name_in_resoures("cloudbuild") then "cloudbuild.tf"]: cloudbuild,
+...
+}
+```  
+
+Assuming that one of the configuration files for a specific environment has the parameter key `cloudbuild` set. 
+These parameters will then be interpreted by the `cloudbuild.jsonnet` template. 
+A file named `cloudbuild.tf.json` will then be compiled using the parameters associated with the `cloudbuild` parameter key. 
+
+It is important to understand that once you have a deeper understanding of Kapitan's capabilities, you can organize these files to a style & logic suitable for your organization.
+
+### templates, docs, scripts
+
+Jinja2 is used to generate documentation that sources information from kapitan's inventory. This enables the ability to have dynamic documentation based on your infrastructure configuration changes.
+In `templates/terraform` you will find `README.md.j2`. This is used to generate a `README.md` template to be utilized by terraform's output module.
+
+The following is what generates the documentation:
+
+```json5
+  data: {
+    template_file: {
+      readme: {
+        template: kap.jinja2_template("templates/terraform/README.md.j2", inv),
+      },
+    },
+  },
+
+  output: {
+    "README.md": {
+      value: "${data.template_file.readme.rendered}",
+      sensitive: true,
+    },
+  },
+```
+
+The function `kap.jinja2_template()` (imported from `kapitan.libjsonnet`) is used to convert and interpreter the `README.md.j2` file into a raw string using the inventory for evaluation logic.
+
+Based on the various parameters jinja2 decides which sections of the readme should be included. 
+When terraform runs it will use the output module to generate your desired `README.md` using information from terraform's state.
+
+Scripts are located in the `scripts` directory. They are compiled using jinja2 as the templating language. An example is as follows:
+
+```shell
+export TF_DATA_DIR=$(realpath -m ${DIR}/../../../.TF_DATA_DIR/{{inventory.parameters.name}}) # Folder for TF initialization (preferable outside of compiled)
+export OUTPUT_DIR=$(realpath -m ${DIR}/../../../output/{{inventory.parameters.name}}) # Folder for storing output files (preferable outside of compiled)
+```
+
+It is good practice to utilize this method to improve integration with various CLI based tools. Scripts help to ensure terraform &
+kapitan can function with your CI/CD systems. It generally depends on your organizational workflows. 
+
+### secrets
+
+Although there are no particular secrets in this instance. It is possible to utilize Kapitan secrets as defined in [secrets management](secrets.md). There is truly no limit to the extensibility of this tool.
```

Comment:
"There is truly no limit to the extensibility of this tool." - Definitely appreciate the comment :) but I think it might be misinterpreted if we add it in the terraform docs? 
Might be better to keep this doc purely about terraform?

Reply:
I felt like I was not doing the secrets section much justice.

I do agree with you on that one though.